test/edu/psu/sweng500/team8/puzzleGenerator/PuzzleGeneratorTests.java

```diff
@@ -27,39 +28,56 @@ public void testsGeneratedSolutionfor4SudokuConstraints() {
 			assertTrue(cellToTest > 0 && cellToTest < 10);
 		}
 		
-		//check for row constraint 
-		for(int j = 0; j < 9; j++)
-		{
-			boolean test = false;
-			for(int i = 0; i<9; i++){
-				int cellToTest = grid.getCell(0, i).getNumber();
-				if(cellToTest == j) test = true;
+		//check for row constraint
+		for(int j = 0; j < 9; j++) {
+			Map<Integer, Boolean> rowCheck = new HashMap<Integer, Boolean>();

```

Comment:
Why use a map? Just keep a set and check if it contains. The boolean value is useless.